corehq/apps/api/es.py

```diff
@@ -701,35 +513,55 @@ def consume_params(self, raw_params):
 ]
 
 
-def es_search(request, domain, reserved_query_params=None):
-    return es_search_by_params(request.GET, domain, reserved_query_params)
+def _validate_and_get_es_filter(search_param):
+    #
+    try:
+        _filter = search_param.pop('filter')
+        # custom use case by 'enveritas' project for Form API
+        date_range = _filter['range']['inserted_at']
+        return {
+            'range': {'inserted_at': date_range}
+        }
+    except KeyError:
+        pass
+    try:
+        # Data export tool passes below structure. validate that it is so
+        _range = None
+        try:
+            _range = _filter['or'][0]['and'][0]['range']['server_modified_on']
+        except KeyError:
+            try:
+                _range = _filter['or'][0]['and'][1]['range']['server_modified_on']
+            except KeyError:
+                pass
+
+        if _range:
+            return XFormServerModifiedParams().consume_params({'server_modified_on': _range})
+        else:
+            raise Http400
+    except (KeyError, AssertionError):
+        raise Http400
```

Comment:
seems weird that this raises a Http400 but the one for `inserted_at` silently passes

Reply:
Just because this is consumed latter.